src/Features/CSharp/Portable/Completion/CompletionProviders/CompletionUtilities.cs

```diff
@@ -91,6 +91,38 @@ internal static bool IsTriggerCharacter(SourceText text, int characterPosition,
             return false;
         }
 
+        /// <summary>
+        /// Tells if we are in positions like this: <c>#nullable $$</c> or <c>#pragma warning $$</c>
+        /// </summary>
+        internal static bool IsCompilerDirectiveTriggerCharacter(SourceText text, int characterPosition)
+        {
+            // First we check if current character is space
+            var ch = text[characterPosition--];
+            if (ch != ' ')
```

Comment:
What about tabs (real `\t` character, not 4 spaces)?

Reply:
Hmmmm, I just checked and yes, tab is a valid separator char. But I am not sure, if should show completions in this case. Usually when a user presses tab (no matter if IDE is configured to translate this tab into spaces or not), he thinks, that he has completed the statement and separates it from another one. Having automatially appearing completions from IDE is like a message "This is absolutelly fine, continue writing like this". I am not sure, if we want to "send such message" to the user here. Also it would introduce inconsistency depending on if tab is replaced with spaces or not. So in general I am not a fan of this proposal, but @CyrusNajmabadi what do you think?